src/collector.ts

```diff
@@ -118,6 +120,53 @@ class ReqDependencyCollector {
 
 }
 
+class NaiveGomodParser {
+    constructor(contents: string) {
+        this.dependencies = NaiveGomodParser.parseDependencies(contents);
+    }
+
+    dependencies: Array<IDependency>;
+
+    static parseDependencies(contents:string): Array<IDependency> {
+        const gomod = contents.split("\n");
+        return gomod.reduce((dependencies, line, index) => {
+            // skip any text after '//'
+            if (line.includes("//")) {
+                line = line.split("//")[0];
+            }
+            const version = semverRegex().exec(line)
+            // Skip lines without version string
+            if (version && version.length > 0) {
+                const parts: Array<string>  = line.trim().split(' ');
+                const pkgName:string = (parts[0] || '').trim();
+                if (pkgName.length > 0) {
+                    const entry: IKeyValueEntry = new KeyValueEntry(pkgName, { line: 0, column: 0 });
+                    entry.value = new Variant(ValueType.String, version[0]);
+                    entry.value_position = { line: index + 1, column: line.indexOf(version[0]) + 1 };
```

Comment:
Also semverRegex().exec(line) seems to be returning the version index as well.

`sem().exec('heloo                1.0.0+beta.gpa')` returns

```
[
  '1.0.0+beta.gpa',
  index: 21,
  input: 'heloo                1.0.0+beta.gpa',
  groups: undefined
]
```

where the index could be used as starting position for version

Reply:
semverRegex().exec(line)  returns only list of version string found in line. That's how the code is working. 
sem() is some other package that gives index and version?

Reply:
No, I tried with `semver-regex` only.